### DISCUSSION TITLE
How to use GetChangeFeedIterator in 3.17-preview

### DISCUSSION THREAD

**Discussion by JaapMosselman** [Azure/azure-cosmos-dotnet-v3#2268] [Q&A] [answered]
Over [here ](https://docs.microsoft.com/en-us/azure/cosmos-db/change-feed-pull-model#consuming-an-entire-containers-changes)I read that HasMoreResults effectively always returns true.

With version 3.11-preview I used an Timer to poll every 5 minutes for changes.
But now with version 3.17-preview I could implement that without Timer, because the while is in fact infinite. And in the catch I could do a Thread.Sleep of 5 minutes.
But I wonder if that is a good idea. I think that will in fact use one thread permanently. So I guess using a Timer is still a good idea when I want to poll at a low frequency?

```
            while (iteratorForTheEntireContainer.HasMoreResults)
            {
                try
                {
                    var configs = await iteratorForTheEntireContainer.ReadNextAsync();
                    _continuationToken = configs.ContinuationToken;

                    foreach (var config in configs)
                    {
                            ......  
                    }
                }
                catch (CosmosException ex) when (ex.StatusCode == System.Net.HttpStatusCode.NotModified)
                {
                    Thread.Sleep(5*60*1000);
                }
            }
```

**Comment by j82w** [comment, marked as answer]
Is there any reason you don't use [Task.Delay instead of the Thread.Sleep](https://stackoverflow.com/questions/20082221/when-to-use-task-delay-when-to-use-thread-sleep) to avoid blocking the thread?

@timsander1 and/or @ealsur any suggestions?

**Reply by ealsur** [reply]
No suggestions on my side other than not blocking the thread, use Task.Delay.

**Reply by JaapMosselman** [reply]
Good suggestion to use Task.Delay. Did not think of that. I simply copied from the samples [here](https://docs.microsoft.com/en-us/azure/cosmos-db/change-feed-pull-model).
Perhaps an idea to add a note to these examples about this?

**Reply by j82w** [reply]
I sent out a PR to update the sample to use Task.Delay instead of Thread.Sleep. Thanks for letting us know.
https://github.com/MicrosoftDocs/azure-docs-pr/pull/148540
